lib/authenticate_methods.rb

```diff
@@ -74,6 +54,54 @@ def authenticate_admin!
 
   protected #################
 
+  def use_signed_params_and_redirect
+    log_out_unlinked_student || log_in_known_LMS_user || prepare_for_new_LMS_user
+  end
+
+  def known_LMS_user
+    @known_LMS_user ||= UserExternalUuid.find_by_uuid(signed_params['uuid'])&.user
+  end
+
+  # If a student is already logged in, but their account is not yet linked with the LMS, log them out.
+  def log_out_unlinked_student
+    if signed_in? && !known_LMS_user && current_user.student?
+      sign_out!(security_log_data: {reason: 'LMS student'})
+      redirect_to(newflow_signup_student_path(request.query_parameters))
+      return true
```

Comment:
the return here is superfluous 

http://ruby-for-beginners.rubymonstas.org/writing_methods/return_values.html

Reply:
Fixed.